downstairs/src/extent.rs

```diff
@@ -653,13 +654,19 @@ pub(crate) fn move_replacement_extent<P: AsRef<Path>>(
     sync_path(&original_file, log)?;
 
     // We distinguish between SQLite-backend and raw-file extents based on the
-    // presence of the `.db` file.  We should never do live migration across
-    // different extent formats; in fact, we should never live-migrate
-    // SQLite-backed extents at all, but must still handle the case of
-    // unfinished migrations.
+    // presence of the `.db` file.  Under normal conditions, we should never
+    // do extent repair across different extent formats; in fact, we should
+    // never extent repair SQLite-backed extents at all, but must still
+    // handle the case of unfinished migrations.
+    // If we are replacing a downstairs, and have created the region empty
+    // with the plan to replace it, then we can have a mismatch between
+    // what files we have and what files we want to replace them with.  When
+    // repairing a downstairs, we don't know in advance if the source of our
+    // repair will be SQLite-backend or not.
```

Comment:
I found this confusing, how about something like
```
    // We distinguish between SQLite-backend and raw-file extents based on the
    // presence of the `.db` file.  We should never do extent repair across
    // different extent formats; it must be SQLite-to-SQLite or raw-to-raw.
    //
    // It is uncommon to perform extent repair on SQLite-backed extents at all,
    // because they are automatically migrated into raw file extents or
    // read-only.  However, it must be supported for two cases:
    //
    // - If there was an unfinished replacement, we must finish that replacement
    //   before migrating from SQLite -> raw file backend, which happens
    //   automatically later in startup.
    // - We use this same code path to perform clones of read-only regions,
    //   which may be SQLite-backed (and will never migrate to raw files,
    //   because they are read only).  This is only the case when the `clone`
    //   argument is `true`.
    //
    // In the first case, we know that we are repairing an SQLite-based extent
    // because the target (original) extent includes a `.db` file.
    //
    // In the second case, the target (original) extent is not present, so we
    // check whether the new files include a `.db` file.
```

Reply:
Done in [7bef555](https://github.com/oxidecomputer/crucible/commit/7bef55560bd1a345855ca77b6c296616ee66e0fe)